File: 15923.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
	int N; cin >> N;

	int start_x, start_y;
	int before_x, before_y;
	cin >> start_x >> start_y;
	before_x = start_x;
	before_y = start_y;
	int distance = 0;
	for (int i = 0; i < N - 1; i++) {
		int x, y; cin >> x >> y;

		distance += sqrt(pow(x - before_x, 2) + pow(y - before_y, 2));
		before_x = x;
		before_y = y;
	}

	distance += sqrt(pow(start_x - before_x, 2) + pow(start_y - before_y, 2));
	cout << distance << endl;
}

File: 15923.py
from math import sqrt

N = int(input())

start_x, start_y = map(int, input().split())
before_x = start_x; before_y = start_y
distance = 0
for i in range(N - 1):
    x, y = map(int, input().split())

    tmp = int(sqrt((x - before_x) ** 2 + (y - before_y) ** 2))
    distance += tmp
    before_x = x; before_y = y

tmp = int(sqrt((start_x - before_x) ** 2 + (start_y - before_y) ** 2))
distance += tmp
print(distance)
